LUMI-O Policies and ACLs

Nortamo opened the pull request:
Adds a new page with more advanced features of the S3 api. 

Mainly examples how you can configure ACLs and policies to restrict object access or share specific data with another project

Done.
LUMI-O admins approve. 

This should probably be squash merged, to keep the history a bit cleaner.

Nortamo committed:
Skeleton for more lumio documentation

hdrei committed:
Update mkdocs.yml

Updating the mkdocs.yml with a page name "Advanced usage of LUMI-O", the name can be changed if you want to call it something else in the side bar

hdrei commented on mkdocs.yml:
```diff
@@ -102,6 +102,7 @@ nav:
       - LUMI-O usage: storage/lumio/index.md
       - Authentication for LUMI-O: storage/lumio/auth-lumidata-eu.md
       - Error messages: storage/lumio/error-messages.md
+      - Advanced usage of LUMI-O: /storage/lumio/advanced.md
```
```suggestion
      - Advanced usage of LUMI-O: storage/lumio/advanced.md
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+1. Policies
+2. Access control list
+
+While ACLs are simpler to configure, they are are an older
```
```suggestion
While ACLs are simpler to configure, they are an older
```

hdrei committed:
Update mkdocs.yml again

There was an extra slash (and I want the rahtiapp preview to work)

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+make objects and buckets public or private. This section
+explains how to apply more granular rules than fully private/public to e.g:
+
+- Share data with a another lumi project.
```
```suggestion
- Share data with another lumi project.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+- Share data with a another lumi project.
+- Restrict object access to specific IP:s
+- Make it a bit harder to accidentally delete data
+- Allow external modification to just specific objects. 
```
```suggestion
- Allow external modification to only specific objects. 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+There are two ways to manage access for data in LUMI-O:
+
+1. Policies
+2. Access control list
```
```suggestion
2. Access control list (ACL)
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+- ACLs can only be used to allow more access, not restrict access from the default
+- ACLs can be applied to to buckets and objects while policies can only be applied to buckets
+	- Though you can create bucket policies which only affect specific objects in the bucket. 
+	- This also means that you will have to individually / recursively apply ACL changes to all objects in a bucket + the bucket itself.
```
```suggestion
Some other differences include:
- ACLs can only be used to allow more access, not restrict access from the default
- ACLs can be applied to buckets and objects while policies can only be applied to buckets
    - Although you can create bucket policies which only affect specific objects in the bucket. 
    - This also means that you will have to individually / recursively apply ACL changes to all objects in a bucket + the bucket itself.
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,161 @@
+You can apply policies to a bucket using `s3cmd` or `aws` commands:
+
+```
+s3cmd setpolicy policy.json  s3://<bucket_name>/
````
```suggestion
s3cmd setpolicy policy.json s3://<bucket_name>/
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,161 @@
+```
+
+This would give `project_<proj_id>` the permission to download
+`myBucket/object1`, but no e.g list other objects in the bucket.
````
```suggestion
`myBucket/object1`, but not e.g list other objects in the bucket.
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,161 @@
+### Configuring Access control lists (ACLs): 
+
+```bash
+s3cmd setacl --recursive --acl-grant=read:'<proj_id>$<proj_id>' s3://<bucket_name>/
````
Is the '<proj_id>$<proj_id>' correct?

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+ACLs  
+
+!!! info
+	The lumi-pub rlcone remotes configured by lumio-conf uses acl settings to make
```
```suggestion
	The lumi-pub rclone remotes configured by lumio-conf uses acl settings to make
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+Would grant read access to all objects in the `<bucket_name>` bucket. 
+The single quotes are important as otherwise the shell might interpret `$<proj_id>` as an (empty) variable
+
+ACLs  
```
Is here coming some list of the different options with ACL?

hdrei replied:
Maybe to add this link? https://s3tools.org/usage

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,161 @@
+
+**s3cmd and rclone**
+
+To access buckets and subsequently objects not owned
+For both `s3cmd` and `rclone` 
```
```suggestion
To access buckets and subsequently objects not owned

For both `s3cmd` and `rclone` 
```

hdrei reviewed (commented):
I see this is a work-in-progress, but I corrected some typos etc

Nortamo committed:
Small example of presigned url

Nortamo committed:
First full draft

Nortamo committed:
Fix typos

Nortamo committed:
Adds explanation on env vars

Kallio commented on docs/storage/lumio/index.md:
I would somehow also add that user should *verify* if the --acl-public is effective properly and for proper objects, since it is rather easy to make a mistake. Even s3cmd supports multiple operations, one should evaluate other tools as well.  (s3cmd is not the best tool, due historical reasons)

Nortamo replied:
Added warning and some more examples for aws so that s3cmd and aws s3api are on the same level.

Nortamo committed:
Adds additional warning and more aws examples

Adds one reminder for the user to actually test
that ACL modifications works, as these can be error prone.

Also adds more aws s3api examples so that it's on the same level
as s3cmd.

Nortamo committed:
Fix bucketname

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+This is not a comprehensive tutorial, more of
+a list of example of the things which are possible using LUMI-O.
+Consult the manual pages of the tools for additional details.
```
```suggestion
This is not a comprehensive tutorial, but more of
a list of examples of things that are possible when using LUMI-O.
Please consult the manual pages of the tools for additional details.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+LUMI-O is an S3 compatible storage solutions. However, this does not mean
+that the system is the same as the "Amazon S3 Cloud Storage". The interface for reading and writing data 
+is exactly the same, but AWS has a bunch of additional features like: self-service provisioning of IAM users,
+Life cycle configuration and write once, read many functionality, which are not really part of "just" s3 storage. 
```
```suggestion
LUMI-O is an S3 compatible storage solution. However, this does not mean
that the system is the same as the "Amazon S3 Cloud Storage". The interface for reading and writing data 
is exactly the same, but AWS has a bunch of additional features, like _self-service provisioning of IAM users_,
_life cycle configuration_ and _write once, read many functionality_, which are not really part of "just" s3 storage. 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+!!! warning 
+	Some advanced  operations which are supported by AWS will complete successfully when run against
+	LUMI-O, e.g object locks, but will actually have no effect. Unless it explicitly stated that a feature
+	is provided by LUMI-O assume that it will not work and be extra thorough in verifying correct functionality. 
```
```suggestion
	LUMI-O, e.g object locks, but will actually have no effect. Unless it is explicitly stated that a feature
	is provided by LUMI-O, assume that it will not work and be extra thorough in verifying correct functionality. 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+### Moving tool configuration files
+
+In some cases it might be required to read
+credentials from some other location then the default 
```
```suggestion
credentials from some other location than the default 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+| FLAG | `--config FILE` | `-c FILE`, `--config=FILE` |                                             |
+
+The `aws` cli additionally has the concept of [profiles](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html), you can specify 
+which one to use using the `--profile <name>` flag or the `AWS_PROFILE` environment variable.
```
```suggestion
The `aws` cli additionally has the concept of [profiles](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html), and you can specify 
which one to use using the `--profile <name>` flag or the `AWS_PROFILE` environment variable.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+Most programs will use the environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
+when trying to authenticate. So these can be set if one does not wish to save the credentials on disk.
+The environment variables do not always take precedent over values set in configuration files, as
```
```suggestion
The environment variables do not always take precedence over values set in configuration files, as
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+when trying to authenticate. So these can be set if one does not wish to save the credentials on disk.
+The environment variables do not always take precedent over values set in configuration files, as
+is the case for `s3cmd` and `rclone`. This means that invalid credentials in config files will
+lead to and access denied even if there are valid credentials in the environment. The `aws` command
```
```suggestion
lead to an access denied even if there are valid credentials in the environment. The `aws` command
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+```python
+import boto3
+
+session = boto3.session.Session(profile_name='lumi-462000439')
````
```suggestion
session = boto3.session.Session(profile_name='lumi-465000001')
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+buckets=s3_client.list_buckets()
+```
+
+Would fetch the buckets and return the information as a python dictionary. 
````
```suggestion
Would fetch the buckets of project 465000001 and return the information as a python dictionary. 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+boto3 uses the same configuration files and respects the same environment variables as the `aws` cli.
+
+!!! note 
+	You will need a sufficently new version (e.g 1.26 which is installed if using python3.6 is to old)  of boto3 for it
+	to understand a default profile set in ~/.aws/credentials and corresponding config file 
+	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
+	profile/endpoint when constructing the client.
```
```suggestion
!!! note 
	You will need a sufficiently new version of boto3 (e.g 1.26 is still fine, which is installed if using python3.6) for it
	to understand a default profile set in ~/.aws/credentials and corresponding config file, 
	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
	profile/endpoint when constructing the client.
```

hdrei replied:
Did I understand it correctly what it tries to say here? The original text was confusing, and trying to clear it out a bit

Nortamo replied:
Fixed in a separate commit

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+Using the rclone config generated by `lumio-conf` or using `s3cmd put -P` you can easily
+make objects and buckets public or private. This section
+explains how to apply more granular rules than fully private/public to e.g:
```
```suggestion
explains how to apply more granular rules than a fully private/public content to e.g:
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+Projects in LUMI-O are single user tenants/accounts where the tenant/account
+names and project names are both the numerical id for the project, 
+i.e 465000001 . 
```
```suggestion
Projects in LUMI-O are handled as "single user tenants/accounts", where the project numerical id (e.g. 465000001) corresponds both the tenant/account name and the project name.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+Subsequently, all members of a LUMI-O project have the exact same 
+rights and permissions, unlike on the LUMI filesystem where files have individual owners.
+Keep this mind if you have critical data in LUMI-O as any other member of your
+LUMI project could accidentally delete it 
```
```suggestion
Subsequently, **all members of a LUMI-O project have the exact same 
rights and permissions**, unlike on the LUMI filesystem, where files have individual owners.
**Keep this mind if you have critical data in LUMI-O as any other member of your
LUMI project could accidentally delete it** 
```

hdrei replied:
Just a suggestion to highlight these lines. The point should somehow be highlighted, I was also considering making another warning-box out of this

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+	Be very carefully when configuring and updating access to buckets and objects.   
+	It's possible to lock yourself out from your own data. Or alternatively make
+	objects visible to the whole world. In the former case, data recovery might not be possible
+	and your data could be permanetly lost.
```
```suggestion
!!! warning
	Be very careful when configuring and updating access to buckets and objects.   
	**It's possible to lock yourself out from your own data**, or alternatively make
	objects visible to the whole world. In the former case, data recovery might not be possible
	and your data could be permanently lost.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+Some other differences include:
+
+- ACLs can only be used to allow more access, not restrict access from the defaults
+- ACLs can be applied to to buckets and objects while policies can only be applied to buckets
```
```suggestion
- ACLs can be applied to buckets and objects while policies can only be applied to buckets
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+```
+
+
+The following example policy would allow the project `444000454` to:
````
```suggestion
The following example policy would allow the project `465000002` to:
```

hdrei replied:
Just a suggestion to use as "an other project" number

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+      "Resource": "arn:aws:s3:::fortheauthenticated/out.json",
+      "Principal": {
+                "AWS": [ 
+                      "arn:aws:iam::444000454:user/444000454"
```
```suggestion
                      "arn:aws:iam::465000002:user/465000002"
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+      "Resource": "arn:aws:s3:::fortheauthenticated",
+      "Principal": {
+                "AWS": [ 
+                      "arn:aws:iam::444000454:user/444000454"
```
```suggestion
                      "arn:aws:iam::465000002:user/465000002"
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+      "Resource": "arn:aws:s3:::fortheauthenticated/upload.json",
+      "Principal": {
+                "AWS": [ 
+                      "arn:aws:iam::444000454:user/444000454"
```
```suggestion
                      "arn:aws:iam::465000002:user/465000002"
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+}
+```
+
+This would allow any one connecting from "lumi-uan04.csc.fi" to upload
````
```suggestion
This would allow any one connecting from "[lumi-uan04.csc.fi](https://docs.lumi-supercomputer.eu/firststeps/loggingin/#lumi-login-nodes-advanced)" to upload
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+For a full list of Actions and resources see the 
+[AWS documentation](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html)
+
+**Don't use action which you do not understand**
```
```suggestion
**Don't use an action which you do not understand**
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+The corresponding operation using `aws s3api`: 
+
+```bash
+aws s3api put-bucket-acl  --acl public-read --bucket <bucket_name>
+aws s3api put-object-acl --acl public-read --bucket <bucket_name> --key <object_name> 
````
Should the first line (281) also be there?

hdrei replied:
Or does the first line specify first the fact that the bucket is readable also (meaning that it is possible to list the content) ...

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+```bash
+aws s3api put-bucket-acl --acl public-read --bucket <bucket_name> 
+```
+Would make the bucket but not the object readable for the world -> Only possible to list the objects
````
```suggestion
Would make the bucket but not the object readable for the world &rarr; Only possible to list the objects
```

hdrei replied:
Please check that the markdown thing that I wrote for "right arrow" is correct and actually produces an arrow ->

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+```bash
+aws s3api put-bucket-acl --bucket fortheauthenticated --acl authenticated-read
+```
+
````
In the above, the "fortheauthenticated" is just a random bucket name, but "authenticated-read" is something that is given as input to --acl ?
Maybe change the "fortheauthenticated" -> "bucket_name" (in < parenthesis, can't be written here correctly)

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+Can be used to grant read-only access to **all** authenticated users of LUMI-O.
+Usefully if data is semi-public but for some reason or another only
+persons with lumi access. Note here that we are only granting read access to the bucket itself
```
```suggestion
Useful if data is semi-public but for some reason or another only
people with lumi access. Note here that we are only granting read access to the bucket itself
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+
+```
+aws s3api get-bucket-acl  --bucket bucket
+aws s3api get-object-acl  --bucket bucket --key object 
````
```suggestion
aws s3api get-bucket-acl  --bucket <bucket_name>
aws s3api get-object-acl  --bucket <bucket_name> --key <object_name> 
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+To view existing ACL:s you can use 
+
+```
+s3cmd info s3://<bucket>/<optional object>
````
```suggestion
s3cmd info s3://<bucket_name>/<optional_object_name>
```

hdrei replied:
Correct?

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+verified -> using invalid credentials will lead to an access denied even for public objects. 
+
+
+Due to the format of the URL, currently there is no know way to use boto3 or `aws` cli
```
```suggestion
Due to the format of the URL, currently there is no known way to use boto3 or `aws` cli
```

hdrei replied:
Was it supposed to read "known" instead?

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+To access buckets and subsequently objects not owned by the authenticated project:
+
+```
+s3cmd ls s3://<proj_id>:bucket/
````
```suggestion
s3cmd ls s3://<proj_id>:<bucket>/
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+```
+
+```
+rclone ls lumi-462000007:"<proj_id>:bucket"
````
```suggestion
rclone ls lumi-465000001:"<proj_id>:<bucket>"
```

hdrei replied:
I was able to confuse myself with if lumi-465000001 (given with rclone) is your own project id, or the project id to which you are trying to access
Maybe to specify this somehow, e.g. below the line
"where lumi-465000001 is your own project, and <project_id> is the project who's bucket you are accessing 
Or something...

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+Don't use curl unless you have to, main
+point here is that the project id has to be included
+with the bucket and object name when generating the signature.
```
```suggestion
Don't use curl unless you have to. But if you do, the main
point here is that the project id has to be included
with the bucket and object name when generating the signature.
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+Don't use curl unless you have to, main
+point here is that the project id has to be included
+with the bucket and object name when generating the signature.
+
```
I'm even more confused with the curl example, which one is your own project id or your own bucket/object (it says "myBucket"), and where do you state the project id, bucket and object that you are trying to fetch/read? 
I guess the things listed in the beginning define the source, but let's then call "myBucket" something else (e.g. Bucket_name) and project "465000002" ?
And "authentication for 465000454" -> "authentication for 465000001" ?

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+You can generate a presigned url using e.g s3cmd 
+
+```bash
+s3cmd signurl s3://example-bucket/file.txt <unix epoch time>
````
```suggestion
s3cmd signurl s3://<bucket_name>/<object_name> <unix_epoch_time>
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+## Presigned URLs
+
+Presigned URLs are URLs generated by the user which grant time-limited "public" access 
+to and object. It's also possible to generate an URL which allows time-limited 
```
```suggestion
to an object. It's also possible to generate an URL which allows time-limited 
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+
+To get the required unix epoch time, it's possible to use online calculators,
+e.g when one wants to grant access until a specific data, or then adding 
+the desired duration to the current time.
```
```suggestion
That generates access link that is valid until the given unix epoch time. To get the required unix epoch time, it's possible to use online calculators,
e.g when one wants to grant access until a specific date, or then adding 
the desired duration to the current time.
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+the desired duration to the current time.
+
+```
+s3cmd signurl s3://example-bucket/file.txt $(echo "`date +%s` + 3600 * <hours>" | bc)
````
```suggestion
s3cmd signurl s3://<bucket_name>/<object_name> $(echo "`date +%s` + 3600 * <nbr_of_hours>" | bc)
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,513 @@
+It's also possible to use the `aws` command to presign:
+
+```bash
+aws s3 presign s3://example-bucket/data.json --expires-in <seconds>
````
```suggestion
aws s3 presign s3://<bucket_name>/<object_name> --expires-in <seconds>
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,513 @@
+### Writable presigned urls
+
+
+There is no way to create presigned urls for `PUT` operations 
+using either `s3cmd` or `aws`, below is a short example script
+using boto3 to generate a valid url.
```
```suggestion
There is no way to create presigned urls for `PUT` operations 
using either `s3cmd` or `aws`. Below is a short example script
using boto3 to generate a valid url that can be then used to add an object called `file.txt` to the defined bucket. 
```

hdrei replied:
Did I understand this correctly???

hdrei reviewed (commented):
Please do check my suggested changes with a thought, since I'm not 100% sure if I did get everything correctly

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Documentation link

Nortamo committed:
Merge branch 'lumio_acl_and_policy' of github.com:Lumi-supercomputer/lumi-userguide into lumio_acl_and_policy

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Merge branch 'lumio_acl_and_policy' of github.com:Lumi-supercomputer/lumi-userguide into lumio_acl_and_policy

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Update docs/storage/lumio/advanced.md

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
More typos

Nortamo committed:
Apply suggestions from code review

Fix more typos

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

Nortamo committed:
Fix merge

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,514 @@
+```
+rclone ls lumi-465000001:"<proj_id>:<bucket>"
+```
+Where `462000007` would be your own project you have configured authentication for
````
```suggestion
Where `465000001` would be your own project you have configured authentication for
```

hdrei commented on docs/storage/lumio/advanced.md:
````diff
@@ -0,0 +1,514 @@
+```
+
+This would allow any one connecting from "[lumi-uan04.csc.fi](https://docs.lumi-supercomputer.eu/firststeps/loggingin/#lumi-login-nodes-advanced)" to upload
+objects starting with `data` to the `almostpublic` bucket (but not download or list them).
````
```suggestion
objects starting with `data` to the bucket called `almostpublic` (but not download or list them).
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,514 @@
+
+The following example policy would allow the project `465000002` to:
+
+- Download the object `out.json` from our bucket `fortheauthenticated`
```
```suggestion
- Download the object `out.json` from our bucket called `fortheauthenticated`
```

hdrei commented on docs/storage/lumio/advanced.md:
```diff
@@ -0,0 +1,514 @@
+boto3 uses the same configuration files and respects the same environment variables as the `aws` cli.
+
+!!! note 
+	You will need a sufficiently new version of boto3 (e.g 1.26 which is installed if using python3.6 is to old) for it
+	to understand a default profile set in ~/.aws/credentials and corresponding config file, 
+	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
+	profile/endpoint when constructing the client.
```
```suggestion
	You will need a sufficiently new version of boto3 (e.g version 1.26, which is installed if using python3.6, is too old) for it
	to understand a default profile set in ~/.aws/credentials and corresponding config file, 
	otherwise the tool will always default to aws s3 endpoint and you will need to specify the
	profile/endpoint when constructing the client.
```

hdrei replied:
Now this is correct interpretation, right? Version 1.26 of boto3 would be too old. (And it's the version that you get if you use python 3.6.)

Nortamo replied:
yes

hdrei committed:
Fix/add some minor things to advanced.md

hdrei reviewed (approved):
Basically this is ready to be merged from my point of view. 
Just one minor change that I'd like someone check, so won't just push the change myself. (Adding basically few commas but depending on if I got the meaning correctly now, they can also be in wrong places...)

Nortamo committed:
Apply suggestions from code review

Co-authored-by: hdrei <78485804+hdrei@users.noreply.github.com>

hdrei commented:
@olouant would you like to check and merge this if you think it's all fine?
